src/Arriba/Arriba.Server/Middlewares/ArribaIdentityMiddleware.cs

```diff
@@ -0,0 +1,85 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.Threading.Tasks;
+using System.Linq;
+using Microsoft.IdentityModel.Tokens;
+using System.Diagnostics;
+using System.Security.Principal;
+using System.Security.Claims;
+using Arriba.Server.Models;
+
+namespace Arriba.Server.Middlewares
+{
+    public class ArribaIdentityMiddleware
```

Comment:
It feels like the bulk of this shouldn't be necessary. I would expect the standard token validation and deserialization to have been done by the existing or pre-build OAuth middleware. Rewriting this adds substantial risk to the code. I would have expected this middleware to take the existing extracted identity, then improve the attributes on it instead of rebuilding the entire thing